electrumsv/network_support/mapi.py

```diff
@@ -218,44 +227,56 @@ def validate_json_envelope(json_response: JSONEnvelope) -> None:
             raise ValueError("MAPI signature invalid")
 
 
-async def broadcast_transaction(tx: "Transaction", server: "NewServer",
-        credential_id: Optional["IndefiniteCredentialId"]) -> None:
-    server_state = server.api_key_state[credential_id]
-    server_state.record_attempt()
+async def broadcast_transaction_mapi_simple(tx: "Transaction", server: "NewServer",
+        credential_id: Optional["IndefiniteCredentialId"], peer_channel: Optional[PeerChannel]=None,
+        merkle_proof: bool=False, ds_check: bool=False) -> BroadcastResponse:
+    server.api_key_state[credential_id].record_attempt()
 
     url = server.url if server.url.endswith("/") else server.url +"/"
     url += "tx"
     # It is unclear if we need to pass false values for these, the specification implies that
     # we should but in theory it won't let us broadcast at all.
     params = {
-        "merkleProof": "false",
-        "dsCheck": "false",
+        'merkleProof': 'false' if not merkle_proof else 'true',
+        'merkleFormat': "TSC",
+        'dsCheck': 'false' if not ds_check else 'true',
     }
-    headers = {}
+    if peer_channel is not None:
+        params.update({
+            'callbackURL': peer_channel.get_callback_url(),
+            'callbackToken': f"Bearer {peer_channel.get_write_token().api_key}",
+            # 'callbackEncryption': None  # Todo: add libsodium encryption
+        })
+    headers = {"Content-Type": "application/octet-stream"}
     headers.update(server.get_authorization_headers(credential_id))
     is_ssl = url.startswith("https")
-
     async with aiohttp.ClientSession() as client:
         async with client.post(url, ssl=is_ssl, headers=headers, params=params,
                 data=tx.to_bytes()) as response:
             try:
                 json_response = await decode_response_body(response)
             except (ClientConnectorError, ConnectionError, OSError, SOCKSError):
                 logger.error("failed connecting to %s", url)
+                raise
             else:
                 if response.status != 200:
-                    logger.error("feeQuote request to %s failed with: status: %s, reason: %s",
+                    logger.error("Broadcast request to %s failed with: status: %s, reason: %s",
                         url, response.status, response.reason)
+                    response.raise_for_status()
```

Comment:
The caller should not care about aiohttp. Consider how this failure is presented to the user, and what useful information needs to be shown and raise `BroadcastFailedException` or a subclass of it.